New Datasource: azurerm_virtual_hub_connection (#21681)
hashicorp/terraform-provider-azurerm: website/docs/d/virtual_hub_connection.html.markdown

---
subcategory: "Network"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_virtual_hub_connection"
description: |-
  Gets information about an existing Virtual Hub Connection
---

# Data Source: azurerm_virtual_hub_connection

Uses this data source to access information about an existing Virtual Hub Connection.

## Virtual Hub Connection Usage

```hcl
data "azurerm_virtual_hub_connection" "example" {
  name                = "example-connection"
  resource_group_name = "example-resources"
  virtual_hub_name    = "example-hub-name"
}

output "virtual_hub_connection_id" {
  value = data.azurerm_virtual_hub_connection.example.id
}
```

## Argument Reference

The following arguments are supported:

> * `name` - The name of the Virtual Hub Connection.

Reviewer: ```suggestion
* `name` - The name of the Connection which should be retrieved.
```

* `resource_group_name` - The Name of the Resource Group where the Virtual Hub Connection exists.

> *  `virtual_hub_name` - The name which should be used for Virtual hub using the connection.

Reviewer: can we make this:

```suggestion
*  `virtual_hub_name` - The name of the Virtual Hub where this Connection exists.
```

> * `resource_group_name` - The Name of the Resource Group where the Virtual Hub Connection exists.
> *  `virtual_hub_name` - The name of the Virtual Hub where this Connection exists.

Reviewer 2: ```suggestion
* `resource_group_name` - The Name of the Resource Group where the Virtual Hub Connection exists.

*  `virtual_hub_name` - The name of the Virtual Hub where this Connection exists.
```
